tareas/tarea-6/ejercicio2/main.js

```diff
@@ -0,0 +1,151 @@
+/*
+TAREA:
+Crear una interfaz que permita agregar ó quitar (botones agregar y quitar) inputs+labels para completar el salario anual de cada integrante de la familia que trabaje.
+Al hacer click en "calcular", mostrar en un elemento pre-existente el mayor salario anual, menor salario anual, salario anual promedio y salario mensual promedio.
+
+Punto bonus: si hay inputs vacíos, ignorarlos en el cálculo (no contarlos como 0).
+*/
+
+let inputCounter = 0;
+
+document.querySelector("#agregar-familiar").onclick = function(e) {
+	e.preventDefault();
+	agregarElemento();
+};
+document.querySelector("#eliminar-familiar").onclick = function(e) {
+	e.preventDefault();
+	eliminarElemento();
+};
+
+document.querySelector("#calcular").onclick = function(e) {
+	e.preventDefault();
+	const paragraphs = document.querySelectorAll(".paragraph-element");
+	paragraphs.forEach(element => {
+		element.remove();
+	});
+	const inputsEdadesCrudo = document.querySelectorAll(".generated-element-input");
+	const inputsEdades = [];
+	for (let elemento of inputsEdadesCrudo) {
+		inputsEdades.push(Number(elemento.value));
+	}
+	const contenedorOutput = document.querySelector("#output");
+	contenedorOutput.appendChild(crearElementoParrafo(calcularSueldoPromedio(inputsEdades), "promedio"));
+	contenedorOutput.appendChild(crearElementoParrafo(calcularSueldoMensualPromedio(inputsEdades), "mensual promedio"));
+	contenedorOutput.appendChild(crearElementoParrafo(calcularSueldoMinimo(inputsEdades), "mínimo"));
+	contenedorOutput.appendChild(crearElementoParrafo(calcularSueldoMaximo(inputsEdades), "máximo"));
+};
+
+document.querySelector("#reset").onclick = function(e) {
+	e.preventDefault();
+	const inputs = document.querySelectorAll(".generated-element-input");
+	inputs.forEach(element => {
+		element.remove();
+	});
+	const labels = document.querySelectorAll(".generated-element-label");
+	labels.forEach(element => {
+		element.remove();
+	});
+	const paragraphs = document.querySelectorAll(".paragraph-element");
+	paragraphs.forEach(element => {
+		element.remove();
+	});
```

Comment:
Si anidaras todos los elementos dentro de un único elemento con clase `grupo`, por ejemplo, podrías sólo remover ese `grupo` y hacer un único for.

Reply:
Corregido